src/time-wait-sync/time-wait-sync.c

```diff
@@ -172,17 +191,45 @@ int main(int argc, char * argv[]) {
                 goto finish;
         }
 
+        r = inotify_init1(IN_NONBLOCK|IN_CLOEXEC);
+        if (r < 0) {
+                log_error_errno(r, "Failed to create inotify descriptor: %m");
+                goto finish;
+        }
+        state.notify_fd = r;
+
+        r = mkdir_safe_label("/run/clock", 0755, 0, 0, MKDIR_WARN_MODE);
+        if (r < 0 && r != -EEXIST) {
+                log_error_errno(r, "Failed to mkdir /run/clock: %m");
+                goto finish;
+        }
+
+        r = inotify_add_watch(state.notify_fd, "/run/clock", IN_CREATE);
+        if (r < 0) {
+                log_error_errno(r, "Failed to watch /run/clock/synchronized: %m");
+                goto finish;
+        }
+
+        r = sd_event_add_io(event, &state.notify_event_source, state.notify_fd,
+                            EPOLLIN, io_handler, &state);
+        if (r < 0) {
+                log_error_errno(r, "Failed to create notify event source: %m");
+                goto finish;
+        }
+
         r = clock_state_update(&state, event);
         if (r > 0) {
                 r = sd_event_loop(event);
                 if (r < 0)
                         log_error_errno(r, "Failed in event loop: %m");
-                else if (state.adjtime_state == TIME_ERROR) {
-                        log_error("Event loop terminated without synchronizing");
-                        r = -ECANCELED;
-                }
         }
 
+        if (state.has_watchfile)
+                log_info("Exit enabled by: /run/clock/synchonized");
+
+        if (state.adjtime_state == TIME_ERROR)
+                log_warning("Exit without adjtimex synchronized.");
```

Comment:
same here, given that this is a pretty common case this definitely doesn't deserve LOG_WARN....

Reply:
I don't know whether it's common.  I do think the message should be at least info, though, so it shows up in the logs if somebody's questioning why the process that waits until synchronization finished without the kernel marking the clock synchronized.  I'll drop it to info.